ksp_plugin/pile_up.cpp

```diff
@@ -424,27 +426,40 @@ DegreesOfFreedom<Barycentric> PileUp::RecomputeFromParts(
   // Then compute the inertia tensor and the angular momentum of the pile-up.
   InertiaTensor<RigidPileUp> pile_up_inertia_tensor;
   Bivector<AngularMomentum, RigidPileUp> pile_up_angular_momentum;
-  RigidTransformation<RigidPileUp, Barycentric> const pile_up_to_barycentric(
-      RigidPileUp::origin,
-      pile_up_barycentre.position(),
-      Identity<RigidPileUp, Barycentric>().Forget());
+  RigidMotion<RigidPileUp, Barycentric> const pile_up_to_barycentric(
+      RigidTransformation<RigidPileUp, Barycentric>(
+          RigidPileUp::origin,
+          pile_up_barycentre.position(),
+          Identity<RigidPileUp, Barycentric>().Forget()),
+      AngularVelocity<Barycentric>{},
+      pile_up_barycentre.velocity());
+  RigidMotion<Barycentric, RigidPileUp> const barycentric_to_pile_up =
+      pile_up_to_barycentric.Inverse();
   for (not_null<Part*> const part : parts) {
-    RigidMotion<RigidPart, Barycentric> const& part_rigid_motion =
+    RigidMotion<RigidPart, Barycentric> const& part_to_barycentric =
         part->rigid_motion();
-    RigidTransformation<RigidPart, Barycentric> const& part_to_barycentric =
-        part_rigid_motion.rigid_transformation();
-    RigidTransformation<RigidPart, RigidPileUp> const part_to_pile_up =
-        pile_up_to_barycentric.Inverse() * part_to_barycentric;
+    RigidMotion<RigidPart, RigidPileUp> const part_to_pile_up =
+        barycentric_to_pile_up * part_to_barycentric;
 
     InertiaTensor<RigidPart> const& part_inertia_tensor =
         part->inertia_tensor();
-    pile_up_inertia_tensor += part->inertia_tensor().Transform(part_to_pile_up);
+    // NOTE(phl): The following call reads like the tensor "transforms" the
+    // rigid motion.  Improve this API.
+    pile_up_inertia_tensor += part->inertia_tensor().Transform(
+        part_to_pile_up.rigid_transformation());
 
     Bivector<AngularMomentum, RigidPart> const part_angular_momentum =
         Anticommutator(part_inertia_tensor,
-                       part_rigid_motion.angular_velocity_of_to_frame());
+                       part_to_barycentric.angular_velocity_of_to_frame());
```

Comment:
`part_to_barycentric.angular_velocity_of_to_frame` is the angular velocity *of* `Barycentric`; that feels wrong somehow.

Reply:
Change to `part_to_pile_up` which is more logical but not different.

Reply:
But still, this is the angular velocity of the pile up axes (i.e., of non rotating axes) in the axes of the part, rather than the angular velocity of the part as seen in non-rotating axes; I think that is a sign error.

Reply:
Fixed by switching everything to `RigidPileUp`.